consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/MergeCoordinator.java

```diff
@@ -77,8 +77,7 @@ public class MergeCoordinator implements MergeMiningCoordinator, BadChainListene
   /** The Merge block creator factory. */
   protected final MergeBlockCreatorFactory mergeBlockCreatorFactory;
   /** The Extra data. */
-  protected final AtomicReference<Bytes> extraData =
```

Comment:
Can't think of a good reason why this needs to be AtomicReference

Reply:
since there is the `setExtraData` method that could change this field after the object has been created, it should be an `AtomicReference` or `volatile`

Reply:
That's fair, but I don't think the MiningCoordinators are actually used by multiple threads in reality? Looks to me like the only actual usage of setExtraData is in JsonBlockImporter which is just a convenience. Might be better to just make this immutable.

Reply:
Think it would be better to leave this as it is using AtomicReference. Whilst the current usage is restricted to JsonBlockImporter it's a public method we can guarantee no one else will call it on a different thread.

As for JsonBlockImporter it's probably not used much as there isn't a JsonBlockExporter but would prefer to not break it. It would be a useful tool to keep.

Reply:
Agree with Jason, until there is a public setter, it should be thread safe.

Reply:
yep, I'm not about to refacotr JsonBlockImporter in this PR, so will use AtomicRef 👍